### PR TITLE
Add fields to Kubernetes application config to allow customizing variant label

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:
I also thought about adding a new field called podselector in [KubernetesTrafficRouting](https://pipecd.dev/docs/user-guide/configuration-reference/#kubernetestrafficrouting), but I did like this PR because it make it to be flexible such as changing the deployment method(ex. quicksync -> canary).

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/config/application_kubernetes.go [hunk 1]
```diff
@@ -34,6 +34,19 @@ type KubernetesApplicationSpec struct {
 	Workloads []K8sResourceReference `json:"workloads"`
 	// Which method should be used for traffic routing.
 	TrafficRouting *KubernetesTrafficRouting `json:"trafficRouting"`
+
+	// The label key of an used by selector.
+	// Default is pipecd.dev/variant.
+	VariantLabelKey string `json:"variantLabelKey"`
```
Comment (nghialv) [inline pkg/config/application_kubernetes.go:40]: Use default tag to set its default value.

```suggestion
	VariantLabelKey string `json:"variantLabelKey" default:"pipecd.dev/variant"`
```

File: pkg/config/application_kubernetes.go [hunk 2]
```diff
@@ -34,6 +34,19 @@ type KubernetesApplicationSpec struct {
 	Workloads []K8sResourceReference `json:"workloads"`
 	// Which method should be used for traffic routing.
 	TrafficRouting *KubernetesTrafficRouting `json:"trafficRouting"`
+
+	// The label key of an used by selector.
```
Comment (nghialv) [inline pkg/config/application_kubernetes.go:38]:
```suggestion
	// The label will be configured to variant manifests used to distinguish them.
```

File: pkg/config/application_kubernetes.go [hunk 3]
```diff
@@ -34,6 +34,19 @@ type KubernetesApplicationSpec struct {
...
+	// Default is pipecd.dev/variant.
+	VariantLabelKey string `json:"variantLabelKey"`
+	// The label value of an used by selector for PRIMARY variant.
```
Comment (nghialv) [inline pkg/config/application_kubernetes.go:41]:
```suggestion
	// The VariantLabelKey value for PRIMARY variant.
```

**Comment by knanao** [comment]
@nghialv 
Thank you for reviewing.
I made some change, so ptal.

**Comment by nghialv** [comment]
Nice. Thank you.
Just a nit about the PR title. How about this

"Add fields to Kubernetes application config to allow customizing variant label"

/lgtm

**Comment by nghialv** [comment]
Working test. 😄 

/lgtm

**Comment by khanhtc1202** [comment]
Here you go
/approve

**Pull request merged**